### ISSUE TITLE
Agent DVR Fails to Connect to Agent

### ISSUE THREAD

**Issue by mtmongoose** [doitandbedone/ispyagentdvr-docker#222]
**Affected Version**
Found at top left "Server Menu" with a server/stack icon. A pop up will be opened, should be in bold letters to the right of your server name. [e.g. v3.0.0.0]

I'm unable to login. I'm using docker image doitandbedone/ispyagentdvr:latest within Docker Swarm.

**Describe the bug**

When accessing http://localhost:8090 I see the spinning wheel and then a failure connecting to agent and am presented with the option to retry.

**Steps To Reproduce**

**Expected behavior**

**Reproduction rate**
How often does this happen in % [e.g. 100% or 50%]

100%

**Screenshots**

![image](https://user-images.githubusercontent.com/32464119/118922177-2a40e780-b8ff-11eb-9252-2e3059cec5b5.png)

**Host System Information (where ispy agent is running/hosted):**
 - OS: [e.g. macOS Catalina 10.15.7, Windows 10, Ubuntu 20.04.1 LTS]

Docker Swarm

**Client System Information (device accessing the web portal, if applicable):**
 - OS: [e.g. macOS Catalina 10.15.7, iOS 8.1, Windows 10, Android 10, Ubuntu 20.04.1 LTS]
 - Browser [e.g. stock browser, safari, chrome]
 - Browser Version [e.g. 86.0.4240.111 (Official Build) (x86_64)]

**Additional Information**
Add any other information about the problem here. This could include configuration files, video/images that may be causing the issue.

version: '3.3'
services:
  app:
    hostname: ispy-local
    environment:
      - NVIDIA_DRIVER_CAPABILITIES=compute,video
      - NVIDIA_VISIBLE_DEVICES=all
    ports:
      - target: 8090
        published: 8090 
        mode: host
      - target: 3478
        published: 3478
        protocol: udp
        mode: host
      - target: 50000
        published: 50000
        protocol: udp
        mode: host
      - target: 50001
        published: 50001
        protocol: udp
        mode: host
      - target: 50002
        published: 50002
        protocol: udp
        mode: host
      - target: 50003
        published: 50003
        protocol: udp
        mode: host
      - target: 50004
        published: 50004
        protocol: udp
        mode: host
      - target: 50005
        published: 50005
        protocol: udp
        mode: host
      - target: 50006
        published: 50006
        protocol: udp
        mode: host
      - target: 50007
        published: 50007
        protocol: udp
        mode: host
      - target: 50008
        published: 50008
        protocol: udp
        mode: host
      - target: 50009
        published: 50009
        protocol: udp
        mode: host
      - target: 50010
        published: 50010
        protocol: udp
        mode: host
    volumes:
      - '/volumes/ispy/config/:/agent/Media/XML/'
      - '/ispy/:/agent/Media/WebServerRoot/Media/'
      - '/volumes/ispy/commands/:/agent/Commands/'
    image: 'doitandbedone/ispyagentdvr:latest'   
    deploy:
      mode: replicated
      replicas: 1
      placement:
         constraints: [node.hostname == xxxxx]
      restart_policy:
        condition: any

**Comment by doitandbedone** [comment]
Which version are you currently using?
Also, could you please attach the logs the printouts from the terminal should do when during the loading message you are seeing.

From what I can see you are using docker compose. It mostly looks correct. Just the ports section seems very different from what I've seen online.

Usually I see something along the lines of:
ports:
  - "8090:8090"
  - "50000-50010:50000-50010/udp"
And for network type:
network_mode: <network name>

If you do use host as network then port forwarding is irrelevant however. Just hopefully you don't have any conflicting ports with other services/containers.

**Comment by ispysoftware** [comment]
Also do try a diffetent browser. Possibly an extension is blocking it and try using the ip address instead of localhost.

**Comment by mtmongoose** [comment]
Sorry for my very late reply. Life has been a little hectic the last few days and then I blew up my Docker Swarm installation while trying to troubleshoot an issue. Still working to straighten a few things out.

@doitandbedone and @ispysoftware thank you both for responding and very quickly. I tried a different browser no luck. I verified that I don't have any conflicting ports. I'm using the latest docker image from Docker Hub. There was no output from the console which makes me think there is an issue starting the app, I'm wondering if the image might be somehow swarm unfriendly. Once I get all my issues worked out, I'll try taking another look at this.
